src/serac/infrastructure/accelerator.hpp

```diff
@@ -75,6 +77,87 @@ enum class ExecutionSpace
  */
 constexpr ExecutionSpace default_execution_space = ExecutionSpace::CPU;
 
+namespace detail {
+
+/**
+ * @brief Trait for "translating" between serac::ExecutionSpace and axom::MemorySpace
+ */
+template <ExecutionSpace space>
+struct execution_to_memory {
+  /// @brief The corresponding axom::MemorySpace
+  static constexpr axom::MemorySpace value = axom::MemorySpace::Dynamic;
+};
```

Comment:
We might be able to just replace the `serac::ExecutionSpace` type in favor of `axom::MemorySpace` too, if there really is just a 1-to-1 mapping between their options

Reply:
For context, here's the definition:
https://github.com/LLNL/axom/blob/e2ed0670e1cd326a07eb00b0ed367fdb5e5edde5/src/axom/core/memory_management.hpp#L36-L46

I think in some sense there isn't a 1-1 mapping because `MemorySpace` allows for multiple kinds of device memory